server/test/integration/vitalsTest/handleHeartbeatTest.js

```diff
@@ -56,6 +61,54 @@ async function lowBatteryHeartbeat(coreId) {
 }
 
 describe('vitals.js integration tests: handleHeartbeat', () => {
+  describe('for a request with an incorrect API key', () => {
+    beforeEach(async () => {
+      await db.clearTables()
+
+      const client = await factories.clientDBFactory(db)
+      await locationDBFactory(db, {
+        locationid: testLocation1Id,
+        radarCoreId: radar_coreID,
+        clientId: client.id,
+      })
+
+      sandbox.stub(braveAlerter, 'startAlertSession')
+      sandbox.stub(braveAlerter, 'sendSingleAlert')
+      sandbox.spy(helpers, 'logSentry')
+      sandbox.stub(db, 'logSensorsVital')
+    })
+
+    afterEach(async () => {
+      await db.clearTables()
+      sandbox.restore()
+    })
```

Comment:
Do you have an understanding of how the `beforeEach` and `afterEach` functions work?

Reply:
i've had experience with them! my user of the `beforeEach` and `afterEach` here were very much based on the other tests  😅

Reply:
Awesome! Just checking that you understood what was going on here :smile: